### PR TITLE
use .moments() in LayerNorm and BatchNorm layers

### PR DESCRIPTION
**Pull request by mikowals**
Mean and variance in the layers are now calculated using `Tensor.moments()`.  I also added tests for both BatchNorm and LayerNorm layers.  

The tests turned up a flaw in how the shape of the `scale` and `offset` which were always of shape `[featureCount]` irrespective of the input shape or axis for normalisation.  That shape leads to incorrect broadcasting when the axis being normalized along is not the last axis.

I have fixed this by always reshaping `scale` and `offset` before they are used.  This seems hacky in that I get the shapes from the calculated `mean` and `variance`.  Without the input shape being known at initialization time though I couldn't see a better way to do this.  

I think the axis argument is probably there to be consistent with Keras but most of the Swift api layers assume inputs and activations are NHWC.  So requiring NHWC, eliminating the axis argument, and the setting the correct shapes in `init()` would be another option.

### CODE REVIEW

File: Sources/TensorFlow/Layers/Normalization.swift [hunk 1]
```diff
@@ -70,48 +70,33 @@ public struct BatchNorm<Scalar: TensorFlowFloatingPoint>: Layer {
...
+        let offsetShape = runningMean.value.shape
+        let inv = rsqrt(runningVariance.value + epsilon) * scale.reshaped(to: scaleShape)
+        return (input - runningMean.value) * inv + offset.reshaped(to: offsetShape)
     }
 
     /// Returns the output obtained from applying the layer to the given input.
     ///
     /// - Parameter input: The input to the layer.
     /// - Returns: The output.
-    @differentiable(vjp: _vjpApplied(to:))
```
Comment (rxwei) [inline Sources/TensorFlow/Layers/Normalization.swift]: Control flow differentiation is known to have memory leaks (#364).  Though it's going to be fixed some time this week, I'd suggest making this change in a later PR instead.

Comment (mikowals) [reply Sources/TensorFlow/Layers/Normalization.swift]: I added back _vjpApplied() as suggested.

Comment (rxwei) [reply Sources/TensorFlow/Layers/Normalization.swift]: Could you please revert the change so that the original indentation won't get modified?

Comment (mikowals) [reply Sources/TensorFlow/Layers/Normalization.swift]: I have reverted to the original for these lines now.

**Comment by BradLarson** [comment]
@mikowals - The inference changes introduced here were triggering shape mismatches within the MiniGo sample project and its tests on tensorflow/swift-models. I've submitted a PR to roll those back here: https://github.com/tensorflow/swift-apis/pull/426 , but if you have a better way to approach this, I'm all ears.
